tools/hive-bigquery/bigquery_component.py

```diff
@@ -0,0 +1,725 @@
+    def check_bq_write_mode(self, mysql_component, hive_table_model,
+                            bq_table_model):
+        """Validates the bq_table_write_mode provided by user
+
+        If the mode is overwrite, drops the tracking table and deletes the
+        BigQuery table. If the mode is create, checks if the tracking table
+        and BigQuery table exist. If the mode is append, checks whether the
+        BigQuery table exists
+
+        Args:
+            mysql_component (:class:`MySQLComponent`): Instance of
+                MySQLComponent to connect to MySQL
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+        """
+
+        if PropertiesReader.get('bq_table_write_mode') == "overwrite":
+            logger.debug("Deleting tracking table and BigQuery table...")
+            mysql_component.drop_table(hive_table_model.tracking_table_name)
+            self.delete_table(bq_table_model.dataset_id,
+                              bq_table_model.table_name)
+            hive_table_model.is_first_run = True
+
+        elif PropertiesReader.get('bq_table_write_mode') == "create":
+            if not hive_table_model.is_first_run:
+                print_and_log("Tracking Table {} already exist".format(
+                    hive_table_model.tracking_table_name), logging.CRITICAL)
+                exit()
+            if self.check_bq_table_exists(bq_table_model.dataset_id,
+                                          bq_table_model.table_name):
+                print_and_log(
+                    "BigQuery Table {} already exist in {} dataset".format(
+                        bq_table_model.table_name, bq_table_model.dataset_id),
+                    logging.CRITICAL)
+
+                exit()
+
+        else:
+            if hive_table_model.is_first_run is False:
+                query = "SELECT COUNT(*) FROM %s WHERE " \
+                        "bq_job_status='RUNNING' OR bq_job_status='DONE'" % \
+                        hive_table_model.tracking_table_name
+                results = mysql_component.execute_query(query)
+                if results[0][0] != 0:
+                    if not self.check_bq_table_exists(bq_table_model.dataset_id,
+                                                      bq_table_model.table_name):
+                        print_and_log(
+                            "Found the tracking table but BigQuery Table {} "
+                            "doesn't exist in {} dataset. Clean up the "
+                            "resources and try again",
+                            logging.CRITICAL)
+                        exit()
+            else:
+                print_and_log(
+                    "Some problem with the tracking table. Check log file for "
+                    "complete details",
+                    logging.CRITICAL)
+                exit()
```

Comment:
I see that we are using exit() to break the program at many places. Is there a way to avoid this at all? Perhaps encapsulate further into smaller functions with appropriate return value.